File: translatorAndroid/src/main/java/com/dev/translatorkmm/android/translate/di/TranslateRepositoryModule.kt
package com.dev.translatorkmm.android.translate.di

import com.dev.translatorkmm.translate.data.local.TranslateLocalDataSource
import com.dev.translatorkmm.translate.data.remote.TranslateRemoteDataSource
import com.dev.translatorkmm.translate.data.repository.TranslateRepositoryImpl
import com.dev.translatorkmm.translate.domain.repository.TranslateRepository
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object TranslateRepositoryModule {

    @Provides
    @Singleton
    fun provideTranslateRepository(
        remoteDataSource: TranslateRemoteDataSource,
        localDataSource: TranslateLocalDataSource
    ): TranslateRepository {
        return TranslateRepositoryImpl(
            remoteDataSource = remoteDataSource,
            localDataSource = localDataSource
        )
    }
}

File: translatorIos/translatorIos/di/translate/TranslateRepositoryModule.swift
import Foundation
import shared

class TranslateRepositoryModule {
    init() {
        @Inject var remoteDataSource: TranslateRemoteDataSource
        @Inject var localDataSource: TranslateLocalDataSource
        
        @Provider var translateRepository: TranslateRepository = TranslateRepositoryImpl(
            remoteDataSource: remoteDataSource,
            localDataSource: localDataSource
        )
    }
}
